warp10/src/main/java/io/warp10/script/WarpScriptTraceableStatement.java

```diff
@@ -0,0 +1,116 @@
+//
+//   Copyright 2023  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script;
+
+/**
+ * Stores a statement, a type, and its position.
+ * This class is json serializable, snapshotable and, is a WarpScriptStackFunction.
+ * Type is defined by {@link WarpScriptTraceableStatement.STATEMENT_TYPE}
+ */
+public class WarpScriptTraceableStatement implements WarpScriptStackFunction {
```

Comment:
Make it explicitely snapshotable rathe than relying on toString

Reply:
The class WarpScriptTraceableStatement lacks companion functions to extract various elements from an instance. As implemented now there are only two ways of using the content of ATTRIBUTE_PARSE_ERRORS (to be renamed), the JSON representation (out of Warp 10), or a dedicated function. We need to provide a way to access the infos in an instance directly from WarpScript without the need for a dedicated function.

Reply:
That's what I did first, but I wanted to have the same output as a standard macro snapshot.
So, I wanted the optimization that prevent 'MYFUNC' FUNCREF EVAL for every functions. But this one works with NamedWarpScriptFunction only, and WarpScriptAuditStatement is obviously not a named function. After some trials in WarpScriptStack.snapshot () and SNAPSHOT.addElement(), this solution does the job without too much refactoring or code duplication.

All the aggregators rely on toString() too, I found it consistent.